src/bdmsupport.c

```diff
@@ -497,6 +517,29 @@ void bdmLaunchGame(item_list_t *itemList, int id, config_set_t *configSet)
     snprintf(bdmCurrentDriver, sizeof(bdmCurrentDriver), "%s", pDeviceData->bdmDriver);
     settings->bdDeviceId = pDeviceData->massDeviceIndex;
 
+    if (!strcmp(bdmCurrentDriver, "ata") && strlen(bdmCurrentDriver) == 3) {
+        // Get DMA settings for ATA mode.
+        int dmaType = 0, dmaMode = 7;
+        configGetInt(configSet, CONFIG_ITEM_DMA, &dmaMode);
+
+        // Set DMA mode and spindown time.
+        if (dmaMode < 3)
+            dmaType = 0x20;
+        else {
+            dmaType = 0x40;
+#ifdef ATA_UDMA_PLUS
+            dmaMode = pDeviceData->ataHighestUDMAMode;
```

Comment:
Chinese IDE-to-SD adapters are faking this value in adapter specs. If the user still can set UDMA mode manually it is ok, but I am not sure if settings for "classic" HDD (HD Loader) will be taken in place in this case. Also some adapters sets this value to zero, while actually supports UDMA4.

Reply:
I'm not sure what to do here about that? Any suggestions?

Reply:
Huh, I am not sure too, maybe read udma mode from HDD apa settings somehow?

Reply:
but that's what it's doing already? on device connection it calls the cmd to retrieve the highest udma mode here https://github.com/ps2homebrew/Open-PS2-Loader/blob/723dedd7a3f6da0bc83859974e58500d3b4f46f1/src/bdmsupport.c#L765
https://github.com/ps2homebrew/Open-PS2-Loader/blob/723dedd7a3f6da0bc83859974e58500d3b4f46f1/modules/hdd/xhdd/xhdd.c#L58

and store the value in `pDeviceData->ataHighestUDMAMode` then at launch its just applying it in this line we're quoting.

Or I'm just misunderstanding your latest comment and you mean read a manually defined udma user setting?

Reply:
> Huh, I am not sure too, maybe read udma mode from HDD apa settings somehow?

Oh wait apa settings.. I didn’t read this properly.. what about this? If ataHighestUDMAMode > 0 use it, otherwise use the user defined setting from the cfg?

Reply:
Yes, I was talking about user defined value that are per game.

Reply:
> Yes, I was talking about user defined value that are per game.

ok now it works very similar to apa hdd, if the automatically resolved udma is <= 0 it will use the user defined value.. we could have it always use the user defined value but then the whole ataHighestUDMAMode is pointless and unused..

so now we check ataHighestUDMAMode is greater than 0 if its not due to a faked value we use the user defined value